mava/wrappers/jumanji.py

```diff
@@ -17,48 +17,107 @@
 import chex
 import jax.numpy as jnp
 from jumanji import specs
+from jumanji.env import Environment
+from jumanji.environments.routing.lbf import LevelBasedForaging
 from jumanji.environments.routing.robot_warehouse import RobotWarehouse
 from jumanji.types import TimeStep
 from jumanji.wrappers import Wrapper
 
 from mava.types import Observation, State
 
 
-class RwareWrapper(Wrapper):
-    """Multi-agent wrapper for the Robotic Warehouse environment."""
-
-    def __init__(self, env: RobotWarehouse):
+class MultiAgentWrapper(Wrapper):
+    def __init__(self, env: Environment):
         super().__init__(env)
-        self._env: RobotWarehouse
+        self._num_agents = self._env.num_agents
+        self.time_limit = self._env.time_limit
 
     def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
-        n_agents = self._env.num_agents
-        observation = Observation(
-            agents_view=timestep.observation.agents_view,
-            action_mask=timestep.observation.action_mask,
-            step_count=jnp.repeat(timestep.observation.step_count, n_agents),
-        )
-        reward = jnp.repeat(timestep.reward, n_agents)
-        discount = jnp.repeat(timestep.discount, n_agents)
-        return timestep.replace(observation=observation, reward=reward, discount=discount)
+        """Modify the timestep for the selected enviroment."""
+        pass
 
     def reset(self, key: chex.PRNGKey) -> Tuple[State, TimeStep]:
-        """Reset the environment. Updates the step count."""
+        """Reset the environment and updates the timestep."""
         state, timestep = self._env.reset(key)
         return state, self.modify_timestep(timestep)
 
     def step(self, state: State, action: chex.Array) -> Tuple[State, TimeStep]:
-        """Step the environment. Updates the step count."""
+        """Step the environment and updates the timestep."""
         state, timestep = self._env.step(state, action)
         return state, self.modify_timestep(timestep)
 
     def observation_spec(self) -> specs.Spec[Observation]:
-        """Specification of the observation of the `RobotWarehouse` environment."""
+        """Specification of the observation of the environment."""
         step_count = specs.BoundedArray(
-            (self._env.num_agents,),
+            (self._num_agents,),
             jnp.int32,
-            [0] * self._env.num_agents,
-            [self._env.time_limit] * self._env.num_agents,
+            [0] * self._num_agents,
+            [self.time_limit] * self._num_agents,
             "step_count",
         )
         return self._env.observation_spec().replace(step_count=step_count)
+
+
+class RWAREWrapper(MultiAgentWrapper):
```

Comment:
I just don't like how it looks with all caps, but I don't mind this too much
```suggestion
class RwareWrapper(MultiAgentWrapper):
```

Reply:
I also back not having all caps